pkg/agent/flowexporter/exporter/exporter.go

```diff
@@ -235,25 +255,55 @@ func (exp *flowExporter) initFlowExporter(collectorAddr string, collectorProto s
 	return nil
 }
 
-func (exp *flowExporter) sendFlowRecords() error {
-	sendAndUpdateFlowRecord := func(key flowexporter.ConnectionKey, record flowexporter.FlowRecord) error {
-		templateID := exp.templateIDv4
+func (exp *flowExporter) sendFlowRecord(dataSetIPv4, dataSetIPv6 ipfix.IPFIXSet) error {
+	addOrUpdateFlowRecord := func(key flowexporter.ConnectionKey, record flowexporter.FlowRecord) error {
 		if record.IsIPv6 {
-			templateID = exp.templateIDv6
-		}
-		dataSet := ipfix.NewSet(ipfixentities.Data, templateID, false)
-		if err := exp.sendDataSet(dataSet, record); err != nil {
-			return err
-		}
-		if err := exp.flowRecords.ValidateAndUpdateStats(key, record); err != nil {
-			return err
+			templateID := exp.templateIDv6
+			// send dataSet when there are two records
+			// TODO: perform message size check before adding record when go-ipfix supports
+			if dataSetIPv6.GetSet().GetNumberOfRecords() >= 2 {
```

Comment:
Looks good. Curious.. isn't `dataSetIPv6.GetSet().GetNumberOfRecords() == 2` sufficient as we add records individually?
Please create a const for `2` record limit.
Please revise the TODO so that it specifies that more than 2 records scenario will be supported in future with changes in go-ipfix.

Reply:
yes it is sufficient. Updated. Thanks!